src/curves.rs

```diff
@@ -760,8 +755,8 @@ macro_rules! new_curve_impl {
 
         impl ConstantTimeEq for $name_affine {
             fn ct_eq(&self, other: &Self) -> Choice {
-                let z1 = self.infinity;
-                let z2 = other.infinity;
+                let z1 = self.is_identity();
+                let z2 = other.is_identity();
 
                 (z1 & z2) | ((!z1) & (!z2) & (self.x.ct_eq(&other.x)) & (self.y.ct_eq(&other.y)))
```

Comment:
Strictly speaking since the identity is always `(0, 0)` anyway, this whole function can just become `self.x.ct_eq(&other.x) & self.y.ct_eq(&other.y)`.

Reply:
Pushed an update.